backend/index.js

```diff
@@ -12,8 +12,19 @@ const jwt = require("jsonwebtoken");
 const notFound = require("./middleware/notfound");
 const BadRequest = require("./errors/badRequest");
 const unAuthenticated = require("./errors/unauthenticated");
-const errorHandlerMiddleware = require("./middleware/errorHandler");
-const uploadMiddleware = multer({ dest: "./uploads" });
+const errorHandlerMiddleware = require("./middleware/errorhandler");
+const storage = multer.diskStorage({
```

Comment:
Let **multer** handle placing uploaded files into the `./uploads` folder instead of doing it ourselves.
Using the date for the filename is not ideal but it'll work for now, a better approach would be to generate a random unique filename (eg: by using the [uuid](https://www.npmjs.com/package/uuid) package)

This code will also fail if you upload a file that doesn't have an extension.